### PR TITLE
Explaining callbacks (with diagrams).

### PR DESCRIPTION
**Pull request by gvwilson**
1.  Fleshing out explanation in `callbacks.md`.
2.  Fixing cross-reference in `basics.md`.
3.  Adding glossary definitions to `gloss.md`.
4.  Create diagrams (all stored in `callbacks.xml`, with parts exports to `callbacks-*.svg`).
5.  Minor changes to variable names in `adder.js` for clarity.

### CODE REVIEW

**Comment by tobyhodges** [comment]
Hey Greg, thanks for your effort on all these figures! They'll take me a little longer to review, as I can't view the SVGs in the GitHub interface. (Unless you know a trick to make that work?)

**Comment by gvwilson** [comment]
If you open the "Files Changed" tab, go down to the offending file, and click on "View File", it's displayed in the browser - tedious, I know, but that's what the Great Hub of Git gives us.

**Comment by tobyhodges** [comment]
Still better than fetch->checkout->open indivudally on my local machine. thanks for the tip!

File: _en/callbacks.md [hunk 1]
```diff
@@ -136,6 +171,15 @@ console.log(`trim then dot: |${trimThenDot}|`)
...
 
+During the call to `temp = first(initial)`,
+but before a value has been returned to be assigned to `temp`,
+memory looks like this:
+
+<figure id="f:callbacks-pipeline">
+  <figcaption>Implementing a Pipeline</figcaption>
+  <img src="../../files/callbacks-pipeline.svg" />
+</figure>
+
```
Comment (tobyhodges) [inline _en/callbacks.md]: In the figure, `first` is pointing to `dot` and `second` to `trim`, but the callbacks provided to these arguments are reversed in the example code. Needs to be fixed wither by changing the example or changing the SVG.

**Pull request merged**